tests/conftest.py

```diff
@@ -238,3 +239,38 @@ def unprivileged_client(
 
         else:
             yield admin_client
+
+
+@pytest.fixture(scope="session")
+def dsc_resource(admin_client: DynamicClient, name: str = "default-dsc"):
+    for dsc in DataScienceCluster.get(dyn_client=admin_client, name=name):
+        yield dsc
+
```

Comment:
```suggestion
    for dsc in DataScienceCluster.get(dyn_client=admin_client, name=name):
        yield dsc
    raise ResourceNotFound....
```
if dsc is not there we should raise